### PR TITLE
(DOCS-14173): Update import Realm app documentation with secrets

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

Updated import and export Realm app documentation to make clear that secrets are not included by default. Also refactored the import steps for clarity and to include how to import secrets. 

### Jira

- https://jira.mongodb.org/browse/DOCS-14173

### Staged Changes (Requires MongoDB Corp SSO)

- [Import an Existing Realm Application](https://docs-mongodborg-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCS-14173/manage-apps/configure/import-realm-app/) (main changes)
- additionally, same note added to the following pages: 
  - https://docs-mongodborg-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCS-14173/manage-apps/configure/export-realm-app/
  - https://docs-mongodborg-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCS-14173/manage-apps/configure/export/export-with-realm-ui/
  - https://docs-mongodborg-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCS-14173/manage-apps/configure/export/export-with-realm-cli/
  - https://docs-mongodborg-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCS-14173/manage-apps/configure/export/export-with-realm-api/

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/includes/steps-import-cli.yaml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  To configure your app with {+cli-bin+}, you must log in to MongoDB Cloud using
+  an :atlas:`API key </configure-api-access/#programmatic-api-keys>` scoped to the organization or
+  project that contains the app.
+
+  .. code-block:: bash
+
+     realm-cli login --api-key="<MongoDB Cloud Public API Key>" --private-api-key="<MongoDB Cloud Private API Key>"
+
+---
+title: Pull the Latest Version of Your Exporting App
```
Comment (dacharyc) [inline source/includes/steps-import-cli.yaml:13]: The wording of this step title doesn't quite make sense to me. Are we instructing the users to pull the latest version of the app they're exporting? 

If that's the case, I would probably reword this to make it clearer. Maybe something like:

> Pull the Latest Version of the App You Want to Import

As this page is about importing an app, focusing on "import" seems more consistent with the purpose of the page.

File: source/includes/steps-import-cli.yaml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+     realm-cli pull --remote="<Your Exported App ID>"
+
+  .. tip::
+
+     You can also download a copy of your application's configuration files from
+     the :guilabel:`Deployment > Export App` screen in the {+ui+}.
+
+---
+title: Create New App
```
Comment (dacharyc) [inline source/includes/steps-import-cli.yaml:29]: This is not a required step when importing an app. You can import an updated configuration into an existing Realm app. Also, the CLI prompts you to create an app if you try to import without an App ID, or import an app ID that doesn't exist in your project. I don't think we need this step here.

File: source/includes/steps-import-cli.yaml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  the exported app in later steps. To create the app, run: 
+
+  .. code-block:: bash
+     
+     realm-cli app create
+
+  Follow the CLI prompts to complete app creation.
+
+---
+title: Copy Exported App Configuration to New App
```
Comment (dacharyc) [inline source/includes/steps-import-cli.yaml:42]: I don't understand this step and the following step. Neither of these is necessary when importing an app, and I'm also not sure what we're trying to get users to do here.

Are we trying to tell users to create a directory where they'll store a configuration for a copy of the exported app? But then where are we telling users to pull down a copy of the new app they just created in the step above (the superfluous step?) I think I'm either mis-understanding what we're trying to instruct users to do, or we're over-complicating this process.

File: source/includes/steps-import-cli.yaml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+     to add to your imported app.
+  #. Gather the secrets. This cannot be done from within {+service-short+}. 
+  #. Add all the secrets to the imported app by following the :ref:`define a secret documentation <define-secret>`.
+     Secrets must be added one at a time.
+
+---
+title: Push New App to Realm
+ref: export-app-cli-push-to-realm
+content: |
+  Now that the app configuration and secrets are updated, push them to your
```
Comment (dacharyc) [inline source/includes/steps-import-cli.yaml:94]: This step feels a little misleading. We're saying that "now that the app configuration and secrets are updated, push _them_ to your remote app." But we're not pushing secrets to the remote app with `realm-cli push` - it's just an updated config. That could be done prior to the step above, or completely independent of it.

File: source/includes/steps-import-github.yaml [hunk 5]
```diff
@@ -0,0 +1,115 @@
+title: Create New App
+ref: export-app-github-create-new-app
+content: |
+  Create a new {+app+}. This is where you'll add the configuration files from
```
Comment (dacharyc) [inline source/includes/steps-import-github.yaml:4]: It isn't required to use the CLI to create the app here, and in fact we might want to point users at creating the app in the UI, since AFAIK you have to [use the UI to configure automated deployment with GitHub](https://docs.mongodb.com/realm/manage-apps/deploy/automated/deploy-automatically-with-github/#specify-a-github-repository).

File: source/includes/steps-import-github.yaml [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+  the exported app in later steps. To create the app, run: 
+
+  .. code-block:: bash
+     
+     realm-cli app create
+
+  Follow the CLI prompts to complete app creation.
+
+---
+title: Set up Github Automatic Deployment
```
Comment (dacharyc) [inline source/includes/steps-import-github.yaml:14]: If we're pointing users at the UI for the above step, I'd probably combine this with the above step. Do all the UI stuff up front in one step.

File: source/includes/steps-import-github.yaml [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+ref: export-app-github-set-up-github
+content: |
+  Follow the :ref:`documentation to set up Github Automatic Deployment <deploy-github>`
+  for your {+app+}.
+
+---
+title: Log In to MongoDB Cloud
+ref: export-app-github-log-in-to-mongodb-cloud
+content: |
+  You need to use the {+cli+} as part of app import process.
```
Comment (dacharyc) [inline source/includes/steps-import-github.yaml:24]: This is not actually true. You could export an app's config from the UI, edit it locally, and then commit and push it to GitHub without ever using the CLI. We actually advise people _not_ to use the CLI when they've configured deploy via GitHub, so it may be best not to use CLI steps in a procedure involving GitHub automated deploy.

It seems like these steps assume a user is making a copy of an existing app. A better procedure for that, if they're using GitHub automated deploy for the existing app, would probably be to clone their existing app config repo into a new directory and make that the config repo for their new app, then edit it as needed and commit it.

**Review by dacharyc — CHANGES_REQUESTED** [review]
Nice work! You've highlighted a gap in our docs here. I see that you've named these includes as replacements for the existing steps on importing a Realm app. These steps, however, seem to assume you're making a copy of a Realm app, and that's not the only time you might import a Realm app. You might also import an updated config into an existing app. So I think the import procedures should stand as they are.

Instead, I think these steps should be an entirely new procedure on a new page called something like "Copy a Realm App".

There are some things we should clarify here - some of these steps aren't really needed as separate steps, and some of them don't make sense to me. But I think this is great start on a new set of procedures!

**Comment by mongodben** [comment]
i think the fundamental question here is how we are defining the term 'import'. 

from your comments and the existing docs, it seems that you're conceptualizing it similarly to [deploy](https://docs.mongodb.com/realm/manage-apps/deploy/manual/deploy-cli/), and the steps file in the current version of the docs is the same as for deploy.

whereas my changes interpret 'import' more as 'migrate configuration to new realm app' (in hindsight this interpretation was largely based on the [commentary in the ticket](https://jira.mongodb.org/browse/DOCS-14173).

File: source/includes/steps-import-cli.yaml (hunk 4 above)
Comment (mongodben) [reply source/includes/steps-import-cli.yaml:94]: good point. updating to say "configuration" instead of "them".

though the secrets should be added before pushing config in case some of the config depends on those secrets existing (such as for auth, for example). if the secrets aren't there, the cli throws an error. 

will add a sentence somewhere noting that.

File: source/includes/steps-import-github.yaml (hunk 5 above)
Comment (mongodben) [reply source/includes/steps-import-github.yaml:4]: fair point. factored to make section method-agnostic

File: source/includes/steps-import-github.yaml (hunk 6 above)
Comment (mongodben) [reply source/includes/steps-import-github.yaml:14]: per previous comment, made above section method-agnostic so no change needed here

File: source/includes/steps-import-github.yaml (hunk 7 above)
Comment (mongodben) [reply source/includes/steps-import-github.yaml:24]: refactored steps to make method agnostic, and to also make export of app as the first step so all the new app stuff is together.

**Comment by mongodben** [comment]
partially implemented changes per suggestion/comments. left a couple of items open pending our discussion this afternoon b/c those sections will be subject to change.

File: source/includes/steps-import-cli.yaml (hunk 1 above)
Comment (mongodben) [reply source/includes/steps-import-cli.yaml]: as part of changes to "copy" page, changing to 'Pull the Latest Version of Your App'

File: source/includes/steps-import-cli.yaml (hunk 3 above)
Comment (mongodben) [reply source/includes/steps-import-cli.yaml]: i think this one now makes sense given the clear context of copying an app on the new page

File: source/includes/note-secrets-not-exported.rst [hunk 8]
```diff
@@ -0,0 +1,4 @@
+.. note:: Secrets Not Exported
+
+   When you export your {+app+}, the export does not include the 
+   :ref:`secrets <app-secret>`.
```
Comment (dacharyc) [inline source/includes/note-secrets-not-exported.rst:4]: Ok, now that we have the new page that spells out the process of copying the secrets, I kind of want this note to link to that procedure. Maybe add a link to the [Migrate Secrets to New App](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCS-14173/manage-apps/configure/copy-realm-app/#migrate-secrets-to-new-app) step on the Copy page?

File: source/includes/steps-copy-cli.yaml [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+  .. code-block:: bash
+
+     realm-cli login --api-key="<MongoDB Cloud Public API Key>" --private-api-key="<MongoDB Cloud Private API Key>"
+
+---
+title: Pull the Latest Version of Your App
+ref: copy-app-cli-pull-the-latest-version-of-your-app
+content: |
+  You'll need a local copy of the configuration files for the {+app+} you're copying.
```
Comment (dacharyc) [inline source/includes/steps-copy-cli.yaml:16]: Per [MongoDB style guidelines](https://docs.mongodb.com/meta/style-guide/writing/use-present-tense/), we should avoid future tense.

```suggestion
  You need a local copy of the configuration files for the {+app+} you're copying.
```

File: source/includes/steps-copy-cli.yaml [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+  .. tip::
+
+     You can also download a copy of your application's configuration files from
+     the :guilabel:`Deployment > Export App` screen in the {+ui+}.
+
+---
+title: Create New App
+ref: copy-app-cli-create-new-app
+content: |
+  Create a new {+app+}. This is where you'll add the configuration files from
```
Comment (dacharyc) [inline source/includes/steps-copy-cli.yaml:32]: Future tense.

```suggestion
  Create a new {+app+}. This is where you add the configuration files from 
```

File: source/includes/steps-copy-cli.yaml [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+
+     realm-cli pull --remote="<Your App ID>"
+
+  .. tip::
+
+     You can also download a copy of your application's configuration files from
+     the :guilabel:`Deployment > Export App` screen in the {+ui+}.
+
+---
+title: Create New App
```
Comment (dacharyc) [inline source/includes/steps-copy-cli.yaml:29]: I still think this step isn't needed on this page. When you push an app configuration, you get an option to create an app if you don't want to push to an existing app.

File: source/includes/steps-copy-cli.yaml [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+
+---
+title: Copy App Configuration to New App
+ref: copy-app-cli-copy-exported-app-to-new-app
+content: |
+  Delete all the configuration files in the new app except for ``realm_config.json``.
+  You'll handle ``realm_config.json`` in the next step. 
+
+  Copy all the :ref:`configuration files and folders <app-configuration>` of your exported app
+  to the imported app folder except for the ``realm_config.json`` file.
```
Comment (dacharyc) [inline source/includes/steps-copy-cli.yaml:50]: Maybe switch `imported` to `new` here given the reframing of this page?

```suggestion
  to the new app folder except for the ``realm_config.json`` file.
```

File: source/includes/steps-copy-cli.yaml [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+content: |
+  In the new app's ``realm_config.json``, replace all the contents except for the 
+  ``"app_id"`` and ``"name"``.
+
+  The new app's ``realm_config.json`` should look something like: 
+
+  .. code-block:: javascript
+
+     {
+       "app_id": "mynewapp-abcde", // don't change
```
Comment (dacharyc) [inline source/includes/steps-copy-cli.yaml:64]: Instead of saying `// don't change` here, maybe we should be more explicit with something like `// Use the new app's ID` here and in ln 65?

File: source/includes/steps-copy-cli.yaml [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+  When you :ref:`export your {+app+} <export-realm-app>`, the export does not include the app's 
+  :ref:`secrets <app-secret>`. As secrets cannot be read once they're defined, 
+  you must also have access to the secrets outside of {+service-short+} to import 
+  them into a {+app+}. If your app doesn't have any secrets, you can skip this step.
+
+  To add the secrets from another Realm app:
+
+  #. Get the names of all the secrets from the exported app by following the 
+     :ref:`view secrets documentation <list-secrets>`.
+  #. Save the names of all the secrets to a secure location. You won't have access
```
Comment (dacharyc) [inline source/includes/steps-copy-cli.yaml:84]: Future tense.

```suggestion
  #. Save the names of all the secrets to a secure location. You can't see 
```

File: source/includes/steps-copy-cli.yaml [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+  :ref:`secrets <app-secret>`. As secrets cannot be read once they're defined, 
+  you must also have access to the secrets outside of {+service-short+} to import 
+  them into a {+app+}. If your app doesn't have any secrets, you can skip this step.
+
+  To add the secrets from another Realm app:
+
+  #. Get the names of all the secrets from the exported app by following the 
+     :ref:`view secrets documentation <list-secrets>`.
+  #. Save the names of all the secrets to a secure location. You won't have access 
+     to the actual secrets, but it'll be useful to have a list of the secret names
```
Comment (dacharyc) [inline source/includes/steps-copy-cli.yaml:85]: Future tense.

```suggestion
     to the actual secrets, but it's useful to have a list of the secret names 
```

File: source/includes/steps-copy-cli.yaml [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+  #. Save the names of all the secrets to a secure location. You won't have access 
+     to the actual secrets, but it'll be useful to have a list of the secret names 
+     to add to your imported app.
+  #. Gather the secrets. This cannot be done from within {+service-short+}. 
+  #. Add all the secrets to the imported app by following the :ref:`define a secret documentation <define-secret>`.
+     Secrets must be added one at a time.
+
+  .. note:: Add secrets before pushing app
+
+     You should add all secrets before puhsing up the copy of the {+app+} because
```
Comment (dacharyc) [inline source/includes/steps-copy-cli.yaml:93]:
```suggestion
     You should add all secrets before pushing up the copy of the {+app+} because 
```

File: source/includes/steps-copy-cli.yaml [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+     to add to your imported app.
+  #. Gather the secrets. This cannot be done from within {+service-short+}. 
+  #. Add all the secrets to the imported app by following the :ref:`define a secret documentation <define-secret>`.
+     Secrets must be added one at a time.
+
+  .. note:: Add secrets before pushing app
+
+     You should add all secrets before puhsing up the copy of the {+app+} because 
+     {+service-short+} can require that the secrets exist to push the configuration.
+     If the secrets don't exist, the {+cli+} will throw an error when you push the
```
Comment (dacharyc) [inline source/includes/steps-copy-cli.yaml:95]: Future tense.

```suggestion
     If the secrets don't exist, the {+cli+} throws an error when you push the 
```

File: source/includes/steps-copy-cli.yaml [hunk 18]
```diff
@@ -0,0 +1,114 @@
...
+     You should add all secrets before puhsing up the copy of the {+app+} because 
+     {+service-short+} can require that the secrets exist to push the configuration.
+     If the secrets don't exist, the {+cli+} will throw an error when you push the 
+     app in the next step.
+
+---
+title: Push New App to Realm
+ref: copy-app-cli-push-to-realm
+content: |
+  Now that the app configuration and secrets are updated, push configuration to your
```
Comment (dacharyc) [inline source/includes/steps-copy-cli.yaml:102]: Reads slightly odd to me without an article.

```suggestion
  Now that the app configuration and secrets are updated, push the configuration to your
```

File: source/includes/steps-copy-cli.yaml [hunk 19]
```diff
@@ -0,0 +1,114 @@
...
+  remote app. {+cli+} immediately deploys the updated configurations on push.
+
+  .. code-block:: bash
+
+     realm-cli push --remote="<Your New App ID>"
+
+  .. tip::
+
+     If you have :ref:`deployment drafts <deployment-draft>` 
+     enabled, you'll be prompted to review and confirm your changes in the
```
Comment (dacharyc) [inline source/includes/steps-copy-cli.yaml:112]: Future tense.

```suggestion
     enabled, you get a prompt to review and confirm your changes in the 
```

File: source/includes/steps-copy-github.yaml [hunk 20]
```diff
@@ -0,0 +1,89 @@
...
+ref: copy-app-github-export-the-latest-version-of-your-app
+content: |
+  Get a local copy of your {+app+} by following the 
+  :ref:`documentation to export your <export-realm-app>`.
+
+---
+title: Create New App
+ref: copy-app-github-create-new-app
+content: |
+  Create a new {+app+}. This is where you'll add the configuration files from
```
Comment (dacharyc) [inline source/includes/steps-copy-github.yaml:11]:
```suggestion
  Create a new {+app+}. This is where you add the configuration files from 
```

File: source/includes/steps-copy-github.yaml [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+  When you :ref:`export your {+app+} <export-realm-app>`, the export does not include the app's 
+  :ref:`secrets <app-secret>`. As secrets cannot be read once they're defined, 
+  you must also have access to the secrets outside of {+service-short+} to import 
+  them into a {+app+}. If your app doesn't have any secrets, you can skip this step.
+
+  To add the secrets from another Realm app:
+
+  #. Get the names of all the secrets from the exported app by following the 
+     :ref:`view secrets documentation <list-secrets>`.
+  #. Save the names of all the secrets to a secure location. You won't have access
```
Comment (dacharyc) [inline source/includes/steps-copy-github.yaml:65]:
```suggestion
  #. Save the names of all the secrets to a secure location. You can't see 
```

File: source/includes/steps-copy-github.yaml [hunk 22]
```diff
@@ -0,0 +1,89 @@
...
+  :ref:`secrets <app-secret>`. As secrets cannot be read once they're defined, 
+  you must also have access to the secrets outside of {+service-short+} to import 
+  them into a {+app+}. If your app doesn't have any secrets, you can skip this step.
+
+  To add the secrets from another Realm app:
+
+  #. Get the names of all the secrets from the exported app by following the 
+     :ref:`view secrets documentation <list-secrets>`.
+  #. Save the names of all the secrets to a secure location. You won't have access 
+     to the actual secrets, but it'll be useful to have a list of the secret names
```
Comment (dacharyc) [inline source/includes/steps-copy-github.yaml:66]:
```suggestion
     to the actual secrets, but it's useful to have a list of the secret names 
```

File: source/includes/steps-copy-github.yaml [hunk 23]
```diff
@@ -0,0 +1,89 @@
...
+
+  .. code-block:: sh
+
+     git add -A
+     git commit -m "<commit message>"
+     git push origin <branch name>
+
+  Once you successfully push your changes to GitHub, {+service-short+}
+  immediately deploys a new version of your application that matches the state
+  of the latest commit. Client applications will automatically use the newest
```
Comment (dacharyc) [inline source/includes/steps-copy-github.yaml:87]:
```suggestion
  of the latest commit. Client applications automatically use the newest
```

**Review by dacharyc — APPROVED** [review]
Just a couple more suggestions, including some future tense stuff to clean up, but it's looking good! Nice work!

File: source/includes/steps-copy-cli.yaml (hunk 14 above)
Comment (mongodben) [reply source/includes/steps-copy-cli.yaml:84]: note to self: not committing this suggestion b.c need to also remove 'to' from the beginning of the next line

File: source/includes/note-secrets-not-exported.rst (hunk 8 above)
Comment (mongodben) [reply source/includes/note-secrets-not-exported.rst:4]: added to note

File: source/includes/steps-copy-cli.yaml (hunk 11 above)
Comment (mongodben) [reply source/includes/steps-copy-cli.yaml:29]: per our convo keeping this as is b.c section necessary pre-req for the step w secrets migration before push
